codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/UserAgentMiddleware.kt

```diff
@@ -22,10 +22,11 @@ class UserAgentMiddleware(val settings: SwiftSettings) : MiddlewareRenderable {
     override val position = MiddlewarePosition.BEFORE
 
     override fun render(writer: SwiftWriter, op: OperationShape, operationStackName: String) {
+        writer.write("let apiMetadata = ${AWSClientRuntimeTypes.Core.APIMetadata}(serviceId: serviceName, version: \"${settings.moduleVersion}\")")
         writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: \$N(${middlewareParamsString()}))", AWSClientRuntimeTypes.Core.UserAgentMiddleware)
     }
 
     private fun middlewareParamsString(): String {
-        return "metadata: ${AWSClientRuntimeTypes.Core.AWSUserAgentMetadata}.fromEnv(apiMetadata: ${AWSClientRuntimeTypes.Core.APIMetadata}(serviceId: serviceName, version: \"${settings.moduleVersion}\"))"
+        return "metadata: ${AWSClientRuntimeTypes.Core.AWSUserAgentMetadata}.fromEnv(apiMetadata: apiMetadata, frameworkMetadata: config.frameworkMetadata)"
```

Comment:
Same nit as above. Can we use $N here for consistency?

Reply:
Hmm i don't think it is possible to do this here because this is returning a string rather than a call to `writer.write()`

Reply:
Fair point. Leave this one alone. My bad for not seeing that